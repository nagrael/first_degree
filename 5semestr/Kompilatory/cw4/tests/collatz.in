# "mathematics is not yet ready for such problems"

int c = 65735346575434578;
int iter = 0;


int step(int c, int div, int mul, int add) {
    if (c % 2 == 0)
        return c / div;
    else
        return mul * c + add;
}

repeat {
    c = step(c, 2, 3, 1);
    iter = iter + 1;
} until c == 1;

print "Reached 1 after";
print iter;
print "iterations";
