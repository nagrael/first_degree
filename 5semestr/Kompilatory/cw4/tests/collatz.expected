Reached 1 after
431
iterations
